src/components/Menu/MenuItem.tsx

```diff
@@ -121,51 +129,71 @@ class MenuItem extends UIComponent<Extendable<IMenuItemProps>, IMenuItemState> {
      */
     renderIcon: PropTypes.func,
 
+    /**
+     * A custom render function the wrapper slot.
+     *
+     * @param {React.ReactType} Component - The computed component for this slot.
+     * @param {object} props - The computed props for this slot.
+     * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+     */
+    renderWrapper: PropTypes.func,
+
     /** Additional CSS styles to apply to the component instance.  */
     styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
 
     /** Override for theme site variables to allow modifications of component styling via themes. */
     variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Shorthand for the wrapper component */
+    wrapper: PropTypes.oneOfType([PropTypes.node, PropTypes.object]),
   }
 
   static defaultProps = {
-    as: 'li',
+    as: 'a',
     accessibility: menuItemBehavior as Accessibility,
+    wrapper: {},
   }
 
   state = IsFromKeyboard.initial
 
   renderComponent({ ElementType, classes, accessibility, rest }) {
-    const { children, content, icon, renderIcon } = this.props
+    const { children, content, icon, renderIcon, renderWrapper, wrapper } = this.props
 
-    return (
+    const menuItemInner = childrenExist(children) ? (
+      children
+    ) : (
       <ElementType
-        className={classes.root}
-        {...accessibility.attributes.root}
-        {...accessibility.keyHandlers.root}
+        className={cx('ui-menu__item__anchor', classes.anchor)}
+        onClick={this.handleClick}
+        onBlur={this.handleBlur}
+        onFocus={this.handleFocus}
+        {...accessibility.attributes.anchor}
+        {...accessibility.keyHandlers.anchor}
         {...rest}
       >
-        {childrenExist(children) ? (
-          children
-        ) : (
-          <a
-            className={cx('ui-menu__item__anchor', classes.anchor)}
-            onClick={this.handleClick}
-            onBlur={this.handleBlur}
-            onFocus={this.handleFocus}
-            {...accessibility.attributes.anchor}
-            {...accessibility.keyHandlers.anchor}
-          >
-            {icon &&
-              Icon.create(this.props.icon, {
-                defaultProps: { xSpacing: !!content ? 'after' : 'none' },
-                render: renderIcon,
-              })}
-            {content}
-          </a>
-        )}
+        {icon &&
+          Icon.create(this.props.icon, {
+            defaultProps: { xSpacing: !!content ? 'after' : 'none' },
+            render: renderIcon,
+          })}
+        {content}
       </ElementType>
     )
+
+    if (wrapper) {
+      return createHTMLListItem(wrapper, {
+        defaultProps: {
+          className: classes.root,
```

Comment:
there are a few problems caused by this; when we pass custom `styles` prop at the `MenuItem` level, they will be applied to the `li` wrapper instead of the `a` link

![screen shot 2018-10-10 at 22 24 53](https://user-images.githubusercontent.com/5442794/46764180-0d21f200-ccdc-11e8-84a1-26e615c99609.png)

It can be fixed by renaming the `root` slot to something like wrapper here and add `classes.root` to the `a` link; here are the changes:

![screen shot 2018-10-10 at 22 39 29](https://user-images.githubusercontent.com/5442794/46764677-62123800-ccdd-11e8-96e2-c51b43c2a28c.png)

and here is the expected result (contains the changes):

![screen shot 2018-10-10 at 22 38 14](https://user-images.githubusercontent.com/5442794/46764691-6f2f2700-ccdd-11e8-8818-28ab0749d5c9.png)

Reply:
problem is the tests will fail now..

Reply:
well we can definitely rename `ui-menu__item` to `ui-menu__item__link` or something similar and use `item` instead of `wrapper` an name that `ui-menu__item` instead of `ui-menu-wrapper`

Reply:
here is another example that shows that `styles` prop and `wrapper.style` are styling the same element, the wrapper and `wrapper.style` with the latter having higher priority since it's applied to the element style directly; currently we cannot target the `a` link to style it:

![screen shot 2018-10-11 at 12 44 55](https://user-images.githubusercontent.com/5442794/46799209-7cd3c380-cd54-11e8-8de0-2cc772111d02.png)